### DISCUSSION TITLE
New parser & lexer?

### DISCUSSION THREAD

**Discussion by JRGalvan** [tursodatabase/turso#587] [General]
Hello! I've been building a transpiler for sql (in rust), and at the end I did a small parser generator that I think is very readable {it is LR(K) so it should have less states that an LR(1)}, I would like to know if rewriting the lexer and the parser is something that is worth doing for Limbo? I am not sure if my parser generator would be more performant than the actual but making the parser more performant and more readable sounds like a good challenge for me :P

**Comment by penberg** [comment]
Hey @JRGalvan, we actually had @jussisaurio do something similar before (https://github.com/tursodatabase/limbo/pull/371). While I am open to having something like that in the tree, I don't think it should be the default one because there will be tons of work to do to get compatibility right. I think a potentially better path is to try to improve the vendored one, eventually forking it hard.

**Comment by penberg** [comment]
We now have a new parser as of https://github.com/tursodatabase/turso/pull/2381. It's not integrated to the core, but it's something we're working on.
